pdns/dnsdist-lua.cc

```diff
@@ -936,12 +936,23 @@ void setupLuaConfig(bool client)
 
   g_lua.writeFunction("setDynBlocksAction", [](DNSAction::Action action) {
       if (!g_configurationDone) {
-        if (action == DNSAction::Action::Drop || action == DNSAction::Action::Refused || action == DNSAction::Action::Truncate) {
-          g_dynBlockAction = action;
-        }
-        else {
-          errlog("Dynamic blocks action can only be Drop, Refused or Truncate!");
-          g_outputBuffer="Dynamic blocks action can only be Drop, Refused or Truncate!\n";
+        switch(action) {
+          case DNSAction::Action::Allow: 
+            g_dynBlockAction = action;
+            break;
+          case DNSAction::Action::Drop:
+            g_dynBlockAction = action;
+            break;
+          case DNSAction::Action::Refused:
+            g_dynBlockAction = action;
+            break;
+          case DNSAction::Action::Truncate:
```

Comment:
I'm not sure the switch is needed there, but perhaps we could at least collapse it into two single blocks since all blocks are the same except for the default, so something like:
```
case DNSAction::Action::Allow:
case DNSAction::Action::Drop:
case DNSAction::Action::Refused:
case DNSAction::Action::Truncate:
  g_dynBlockAction = action;
  break;
default:
...
```

Reply:
Ack, will collapse the switch statement. 

With regards to the rule processing, how about permitting the use of DNSAction::Action::None here?

Reply:
Right, it looks like `DNSAction::Action::None` would work for `setDynBlocksAction()`, although it wouldn't for the optional action parameter of `addDynBlocks()` since we already use `None` there to indicate that the default action should be used. I don't think that's an issue, but we would need to make that clear in the documentation.